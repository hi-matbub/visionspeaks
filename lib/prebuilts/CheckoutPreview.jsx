import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import { formatUSD } from '../../utils/formatUSD';

const CheckoutPreview = ({
  cart,
  handleEmptyCartNotice,
  handleUpdatedCartInState,
}) => {
  const [cartInStorage, setCartInStorage] = useState(cart);

  const handleRemoveFromCart = (id) => {
    const updatedCart = cartInStorage;

    delete updatedCart[id];

    setCartInStorage(updatedCart);

    if (Object.keys(updatedCart).length === 0) {
      localStorage.removeItem('cart');
      return handleEmptyCartNotice();
      // eslint-disable-next-line no-else-return
    } else {
      // func requires JSON
      handleUpdatedCartInState(JSON.stringify(updatedCart));
      return localStorage.setItem('cart', JSON.stringify(cartInStorage));
    }
  };

  console.log(cart);
  return (
    <Container>
      <Row className="d-flex align-items-center">
        <h3 className="ellipse pb-5">In Cart</h3>
      </Row>

      {Object.keys(cart).map((i) => (
        <Row key={cart[i].id} className="d-flex align-items-center rounded">
          <Col xs={2}>
            <img
              src={cart[i].images[0]}
              alt={cart[i].name}
              style={{ height: '75px', width: '75px' }}
            />
          </Col>

          <Col xs={10} className="d-flex flex-column">
            <Row className="align-self-end mr-2">
              <h4 style={{ width: '100%', margin: '0' }} className="ellipse">
                {cart[i].name}
              </h4>
            </Row>

            <Row className="align-self-end align-items-end mr-2">
              <p className="mt-2" style={{ margin: '0', padding: '0' }}>
                ${formatUSD(cart[i].price)} USD
              </p>
              <p className="ml-4 mr-2" style={{ margin: '0', padding: '0' }}>
                Qty: {cart[i].qty}
              </p>
              <button
                type="button"
                style={{
                  border: 'none',
                  background: 'none',
                  margin: '0',
                  padding: '0',
                }}
                className="btn btn-outline-danger font-weight-bolder ml-4"
                onClick={() => {
                  handleRemoveFromCart(i);
                }}
              >
                X
              </button>
            </Row>
          </Col>
        </Row>
      ))}
    </Container>
  );
};

CheckoutPreview.propTypes = {
  cart: PropTypes.shape({
    images: PropTypes.string,
    name: PropTypes.string,
    price: PropTypes.string,
    qty: PropTypes.string,
  }),
  handleEmptyCartNotice: PropTypes.func.isRequired,
  handleUpdatedCartInState: PropTypes.func.isRequired,
};

CheckoutPreview.defaultProps = {
  cart: {
    images: '',
    name: '',
    price: '',
    qty: '',
  },
};

export default CheckoutPreview;
