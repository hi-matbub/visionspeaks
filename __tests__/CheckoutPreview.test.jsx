import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CheckoutPreview from '../lib/prebuilts/CheckoutPreview';

const cart = {
  one: {
    name: 'Strawberry Bone',
    description: 'Super colorful, your dog will love this!',
    price: 197,
    id: 'srw_bn_',
    images: ['/pink.png'],
  
  }
};

const handleEmptyCartNotice = () => {}
const handleUpdatedCartInState = () => {}

it('CheckoutPreview component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <CheckoutPreview 
      cart={cart}  
      handleEmptyCartNotice={handleEmptyCartNotice}
      handleUpdatedCartInState={handleUpdatedCartInState}
    />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('CheckoutPreview component renders unchanged', () => {
  const tree = renderer.create(
    <CheckoutPreview 
      cart={cart}  
      handleEmptyCartNotice={handleEmptyCartNotice}
      handleUpdatedCartInState={handleUpdatedCartInState}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
