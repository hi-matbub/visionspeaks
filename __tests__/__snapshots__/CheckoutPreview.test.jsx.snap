// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutPreview component renders unchanged 1`] = `
<div
  className="container"
>
  <div
    className="d-flex align-items-center row"
  >
    <h3
      className="ellipse pb-5"
    >
      In Cart
    </h3>
  </div>
  <div
    className="d-flex align-items-center rounded row"
  >
    <div
      className="col-2"
    >
      <img
        alt="Strawberry Bone"
        src="/pink.png"
        style={
          Object {
            "height": "75px",
            "width": "75px",
          }
        }
      />
    </div>
    <div
      className="d-flex flex-column col-10"
    >
      <div
        className="align-self-end mr-2 row"
      >
        <h4
          className="ellipse"
          style={
            Object {
              "margin": "0",
              "width": "100%",
            }
          }
        >
          Strawberry Bone
        </h4>
      </div>
      <div
        className="align-self-end align-items-end mr-2 row"
      >
        <p
          className="mt-2"
          style={
            Object {
              "margin": "0",
              "padding": "0",
            }
          }
        >
          $
          1.97
           USD
        </p>
        <p
          className="ml-4 mr-2"
          style={
            Object {
              "margin": "0",
              "padding": "0",
            }
          }
        >
          Qty: 
        </p>
        <button
          className="btn btn-outline-danger font-weight-bolder ml-4"
          onClick={[Function]}
          style={
            Object {
              "background": "none",
              "border": "none",
              "margin": "0",
              "padding": "0",
            }
          }
          type="button"
        >
          X
        </button>
      </div>
    </div>
  </div>
</div>
`;
